src/query/storage/m3/storage.go

```diff
@@ -197,6 +200,13 @@ func (s *m3storage) FetchCompressed(
 		return result, noop, err
 	}
 
+	if processor := queryOptions.IterationOptions.SeriesIteratorProcessor; processor != nil {
+		if err := processor.InspectSeries(ctx, result.SeriesIterators()); err != nil {
```

Comment:
Inspecting here should probably be async and not block query execution (although this makes series lifecycles more difficult to manage)

Reply:
Made this async and modifed the cleanup func so we ensure inspection finishes before we call `accumulator.Close`